diff --git a/apps/nextjs-website/src/_contents/translations.ts b/apps/nextjs-website/src/_contents/translations.ts
index fac28c4..0234bb7 100644
--- a/apps/nextjs-website/src/_contents/translations.ts
+++ b/apps/nextjs-website/src/_contents/translations.ts
@@ -341,10 +341,20 @@ export const translations = {
       whyCreateAccount: 'Perché iscriversi a PagoPA DevPortal',
       passwordPolicy:
         'Minimo 8 caratteri, almeno un numero, almeno una lettera maiuscola e almeno un carattere speciale',
+      emailSent: (email: string) => `Email inviata a ${email}`,
       advantages: [
-        'Lorem ipsum dolor sit amet, consectetur adipiscing elit, sed do eiusmod tempor incididunt ut labore et dolore magna aliqua.',
-        'Lorem ipsum dolor sit amet, consectetur adipiscing elit, sed do eiusmod tempor incididunt ut labore et dolore magna aliqua.',
-        'Lorem ipsum dolor sit amet, consectetur adipiscing elit, sed do eiusmod tempor incididunt ut labore et dolore magna aliqua.',
+        {
+          title: 'vantaggio 1',
+          text: 'Lorem ipsum dolor sit amet, consectetur adipiscing elit, sed do eiusmod tempor incididunt ut labore et dolore magna aliqua.',
+        },
+        {
+          title: 'vantaggio 2',
+          text: 'Lorem ipsum dolor sit amet, consectetur adipiscing elit, sed do eiusmod tempor incididunt ut labore et dolore magna aliqua.',
+        },
+        {
+          title: 'vantaggio 3',
+          text: 'Lorem ipsum dolor sit amet, consectetur adipiscing elit, sed do eiusmod tempor incididunt ut labore et dolore magna aliqua.',
+        },
       ],
     },
     confirmSignUp: {
diff --git a/apps/nextjs-website/src/app/auth/sign-up/page.tsx b/apps/nextjs-website/src/app/auth/sign-up/page.tsx
index 77c2f93..6cf9f39 100644
--- a/apps/nextjs-website/src/app/auth/sign-up/page.tsx
+++ b/apps/nextjs-website/src/app/auth/sign-up/page.tsx
@@ -4,7 +4,6 @@ import PageNotFound from '@/app/not-found';
 import CheckItem from '@/components/molecules/CheckItem/CheckItem';
 import ConfirmSignUp from '@/components/organisms/Auth/ConfirmSignUp';
 import SignUpForm from '@/components/organisms/Auth/SignUpForm';
-import { environment } from '@/config';
 import { SignUpSteps } from '@/lib/types/signUpSteps';
 import {
   Alert,
@@ -17,6 +16,12 @@ import {
 import { Auth } from 'aws-amplify';
 import { useRouter, useSearchParams } from 'next/navigation';
 import { useCallback, useState } from 'react';
+import { isProduction } from '@/config';
+
+interface Info {
+  message: string;
+  isError: boolean;
+}
 
 const SignUp = () => {
   const {
@@ -37,7 +42,8 @@ const SignUp = () => {
   const [signUpStep, setSignUpStep] = useState(
     params.get('step') || SignUpSteps.SIGN_UP
   );
-  const [error, setError] = useState<string | null>(null);
+
+  const [info, setInfo] = useState<Info | null>(null);
 
   const onSignUp = useCallback(async () => {
     const result = await Auth.signUp({
@@ -52,7 +58,7 @@ const SignUp = () => {
         'custom:company_type': company,
       },
     }).catch((error) => {
-      setError(error.message);
+      setInfo({ message: error.message, isError: true });
       return false;
     });
 
@@ -86,10 +92,14 @@ const SignUp = () => {
 
   const onResendEmail = useCallback(async () => {
     await Auth.resendSignUp(username);
+    setInfo({
+      message: signUp.emailSent(username),
+      isError: false,
+    });
     return null;
-  }, [username]);
+  }, [signUp, username]);
 
-  if (environment === 'prod') {
+  if (isProduction) {
     return <PageNotFound />;
   }
 
@@ -123,8 +133,8 @@ const SignUp = () => {
               return (
                 <CheckItem
                   key={index}
-                  title={`Vantaggio ${index}`}
-                  description={advantage}
+                  title={advantage.title}
+                  description={advantage.text}
                 />
               );
             })}
@@ -162,11 +172,13 @@ const SignUp = () => {
         </Grid>
       </Box>
       <Snackbar
-        open={!!error}
-        autoHideDuration={2000}
-        onClose={() => setError(null)}
+        open={!!info}
+        autoHideDuration={4000}
+        onClose={() => setInfo(null)}
       >
-        <Alert severity='error'>{error}</Alert>
+        <Alert severity={info?.isError ? 'error' : 'success'}>
+          {info?.message}
+        </Alert>
       </Snackbar>
     </>
   );
diff --git a/apps/nextjs-website/src/components/organisms/Auth/SignUpForm.tsx b/apps/nextjs-website/src/components/organisms/Auth/SignUpForm.tsx
index 22f98da..bcff3f7 100644
--- a/apps/nextjs-website/src/components/organisms/Auth/SignUpForm.tsx
+++ b/apps/nextjs-website/src/components/organisms/Auth/SignUpForm.tsx
@@ -85,7 +85,6 @@ const SignUpForm = ({
   } = translations;
 
   const [showPassword, setShowPassword] = useState(false);
-  const [showConfirmPassword, setShowConfirmPassword] = useState(false);
   const [isPasswordValid, setIsPasswordValid] = useState(false);
   const [isPasswordDirty, setIsPasswordDirty] = useState(false);
 
@@ -97,9 +96,6 @@ const SignUpForm = ({
     event.preventDefault();
   };
 
-  const handleClickShowConfirmPassword = () =>
-    setShowConfirmPassword((show) => !show);
-
   const handleMouseDownConfirmPassword = (
     event: MouseEvent<HTMLButtonElement>
   ) => {
@@ -123,16 +119,9 @@ const SignUpForm = ({
 
   const onSignUpClick = useCallback(() => {
     if (
-      !firstName ||
-      firstName?.trim().length === 0 ||
-      !lastName ||
-      lastName?.trim().length === 0 ||
-      !username ||
-      username?.trim().length === 0 ||
-      !password ||
-      password?.trim().length === 0 ||
-      !confirmPassword ||
-      confirmPassword?.trim().length === 0
+      [firstName, lastName, username, password, confirmPassword].some(
+        (value) => !value || value.trim().length === 0
+      )
     ) {
       return;
     }
@@ -167,6 +156,7 @@ const SignUpForm = ({
                     value={firstName}
                     onChange={({ target: { value } }) => setFirstName(value)}
                     helperText={shared.requiredFieldError}
+                    required
                   />
                 </Grid>
                 <Grid item xs={6}>
@@ -175,6 +165,7 @@ const SignUpForm = ({
                     value={lastName}
                     onChange={({ target: { value } }) => setLastName(value)}
                     helperText={shared.requiredFieldError}
+                    required
                   />
                 </Grid>
               </Grid>
@@ -187,6 +178,7 @@ const SignUpForm = ({
                   }}
                   helperText={shared.emailFieldError}
                   customValidators={emailValidators}
+                  required
                 />
               </Stack>
               <Stack spacing={2} mb={2}>
@@ -243,11 +235,11 @@ const SignUpForm = ({
                       <InputAdornment position='end'>
                         <IconButton
                           aria-label='toggle confirm password visibility'
-                          onClick={handleClickShowConfirmPassword}
+                          onClick={handleClickShowPassword}
                           onMouseDown={handleMouseDownConfirmPassword}
                           edge='end'
                         >
-                          {showConfirmPassword ? (
+                          {showPassword ? (
                             <VisibilityOff />
                           ) : (
                             <Visibility />
@@ -268,10 +260,11 @@ const SignUpForm = ({
               </Stack>
               <Stack spacing={2} mb={2}>
                 <FormControl fullWidth>
-                  <InputLabel id='company-field' sx={{ top: '-8px' }}>
+                  <InputLabel id='company-field' sx={{ top: '-8px' }} disabled>
                     {shared.company}
                   </InputLabel>
                   <Select
+                    disabled
                     labelId='company-field'
                     id='company-field-select'
                     value={company}
